core/templates/pages/teach-page/teach-page.component.spec.ts

```diff
@@ -15,112 +15,198 @@
+    expect(component.activeTabName).toBe('teach');
   });
 
-  it('should click on participation tab', function(done) {
-    ctrl.$onInit();
-    expect(ctrl.activeTabName).toBe('teach');
+  it('should click on participation tab', (() => {
+    component.ngOnInit();
+    expect(component.activeTabName).toBe('teach');
 
-    ctrl.onTabClick('participation');
-
-    // Function setTimeout is being used here in order to wait onhashchange
-    // event to finish. setTimeout is executed only after call stack is empty.
-    // Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/EventLoop
-    setTimeout(function() {
-      expect(windowRef.nativeWindow.location.hash).toBe('#participation');
-      expect(ctrl.activeTabName).toBe('participation');
-      done();
-    });
-  });
+    component.onTabClick('participation');
+    expect(windowRef.nativeWindow.location.hash).toBe('#participation');
+    expect(component.activeTabName).toBe('participation');
+  }));
 
-  it('should activate teach tab on init', function() {
+  it('should activate teach tab on init', () => {
     windowRef.nativeWindow.location.hash = '#teach';
 
-    ctrl.$onInit();
+    component.ngOnInit();
 
     expect(windowRef.nativeWindow.location.hash).toBe('#teach');
-    expect(ctrl.activeTabName).toBe('teach');
+    expect(component.activeTabName).toBe('teach');
   });
 
-  it('should activate participation tab on init', function() {
+  it('should activate participation tab on init', () => {
     windowRef.nativeWindow.location.hash = '#participation';
 
-    ctrl.$onInit();
+    component.ngOnInit();
 
     expect(windowRef.nativeWindow.location.hash).toBe('#participation');
-    expect(ctrl.activeTabName).toBe('participation');
+    expect(component.activeTabName).toBe('participation');
   });
 
-  it('should get static image url', function() {
-    expect(ctrl.getStaticImageUrl('/path/to/image')).toBe(
+  it('should get static image url', () => {
+    expect(component.getStaticImageUrl('/path/to/image')).toBe(
       '/assets/images/path/to/image');
   });
 
-  it('should apply to teach with oppia', function() {
-    var applyToTeachWithOppiaEventSpy = spyOn(
-      SiteAnalyticsService, 'registerApplyToTeachWithOppiaEvent')
+  it('should apply to teach with oppia', () => {
+    const applyToTeachWithOppiaEventSpy = spyOn(
+      siteAnalyticsService, 'registerApplyToTeachWithOppiaEvent')
       .and.callThrough();
 
-    ctrl.$onInit();
+    component.ngOnInit();
     spyOnProperty(windowRef, 'nativeWindow').and.returnValue({
-      location: ''
+      location: {
+        href: ''
+      }
     });
-    ctrl.onApplyToTeachWithOppia('/login');
-    $timeout.flush(150);
-
-    expect(windowRef.nativeWindow.location).toBe(
-      'https://goo.gl/forms/0p3Axuw5tLjTfiri1');
-    expect(applyToTeachWithOppiaEventSpy).toHaveBeenCalled();
+    component.onApplyToTeachWithOppia();
+    // Increased time to 300ms to avoid race condition.
```

Comment:
This comment makes sense if we code change now. I don't think comment would mean anything once this goes in `Increased time to 300ms to avoid race condition`.

Please remove it if it makes sense to you.

Reply:
I only wanted to leave it so that if something becomes flaky with this frontend test, the reason is clear. It does make sense to me though, so I've removed it